src/main/scala/com/databricks/labs/overwatch/MultiWorkspaceDeployment.scala

```diff
@@ -239,6 +246,9 @@ class MultiWorkspaceDeployment extends SparkSessionWrapper {
           fullMsg,
           Some(multiWorkspaceParams.deploymentId)
         ))
+    }finally {
+        SparkSessionWrapper.sessionsMap.remove(Thread.currentThread().getId)
+        logger.log(Level.INFO, s"""Removed ${Thread.currentThread().getId} from sessionMap""")
```

Comment:
Instead of triplicating this logic, please create a function that takes in the threadID and completes this logic. This func should probably reside inside of SparkSessionWrapper.